### PR TITLE
Cosmos DB Mongo RBAC API release

### PR DESCRIPTION
**Pull request by ashwinisingh01**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 1]
```diff
@@ -351,6 +353,12 @@ public partial class CosmosDBAccountResource : Azure.ResourceManager.ArmResource
         public virtual Azure.Response<Azure.ResourceManager.CosmosDB.MongoDBDatabaseResource> GetMongoDBDatabase(string databaseName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.CosmosDB.MongoDBDatabaseResource>> GetMongoDBDatabaseAsync(string databaseName, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.ResourceManager.CosmosDB.MongoDBDatabaseCollection GetMongoDBDatabases() { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.CosmosDB.MongoRoleDefinitionGetResultResource> GetMongoRoleDefinitionGetResult(string mongoRoleDefinitionId, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:356]: Rename `MongoRoleDefinitionGetResults` to `MongoDBRoleDefinition` in `rename-mapping` section of `autorest.md`.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 2]
```diff
@@ -351,6 +353,12 @@ public partial class CosmosDBAccountResource : Azure.ResourceManager.ArmResource
...
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.CosmosDB.MongoRoleDefinitionGetResultResource>> GetMongoRoleDefinitionGetResultAsync(string mongoRoleDefinitionId, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.ResourceManager.CosmosDB.MongoRoleDefinitionGetResultCollection GetMongoRoleDefinitionGetResults() { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.CosmosDB.MongoUserDefinitionGetResultResource> GetMongoUserDefinitionGetResult(string mongoUserDefinitionId, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:359]: Rename `MongoUserDefinitionGetResults` to `MongoDBUserDefinition`.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 3]
```diff
@@ -2795,6 +2894,30 @@ public partial class MongoDBIndexConfig
...
+    public partial class MongoRoleDefinitionGetResultCreateOrUpdateContent
+    {
+        public MongoRoleDefinitionGetResultCreateOrUpdateContent() { }
+        public string DatabaseName { get { throw null; } set { } }
+        public Azure.ResourceManager.CosmosDB.Models.MongoRoleDefinitionType? DefinitionType { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.CosmosDB.Models.Privilege> Privileges { get { throw null; } }
+        public string RoleName { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.CosmosDB.Models.Role> Roles { get { throw null; } }
+    }
+    public enum MongoRoleDefinitionType
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2906]: Rename to `MongoDBRoleDefinitionType` to keep consistent.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 4]
```diff
@@ -1220,6 +1230,83 @@ public partial class MongoDBDatabaseThroughputSettingResource : Azure.ResourceMa
...
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
+    }
+    public partial class MongoRoleDefinitionGetResultData : Azure.ResourceManager.Models.ResourceData
+    {
+        public MongoRoleDefinitionGetResultData() { }
+        public string DatabaseName { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.CosmosDB.Models.Privilege> Privileges { get { throw null; } }
+        public string RoleName { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.CosmosDB.Models.Role> Roles { get { throw null; } }
+        public Azure.ResourceManager.CosmosDB.Models.MongoRoleDefinitionType? TypePropertiesType { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:1255]: Add `MongoRoleDefinitionGetResults.properties.type: RoleDefinitionType` in `rename-mapping` section of `autorest.md` to rename this improper auto-renamed property name due to duplication with properties of base class `ResourceData`.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 5]
```diff
@@ -2835,6 +2958,18 @@ public partial class PeriodicModeProperties
         public int? BackupRetentionIntervalInHours { get { throw null; } set { } }
         public Azure.ResourceManager.CosmosDB.Models.CosmosDBBackupStorageRedundancy? BackupStorageRedundancy { get { throw null; } set { } }
     }
+    public partial class Privilege
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2961]: Avoid single-word class name. Rename to a more descriptive name.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 6]
```diff
@@ -2835,6 +2958,18 @@ public partial class PeriodicModeProperties
...
+    {
+        public Privilege() { }
+        public System.Collections.Generic.IList<string> Actions { get { throw null; } }
+        public Azure.ResourceManager.CosmosDB.Models.PrivilegeResource Resource { get { throw null; } set { } }
+    }
+    public partial class PrivilegeResource
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:2967]: Rename to `PrivilegeResourceInfo` as `Resource` suffix has a special meaning in .NET Azure SDK.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs [hunk 7]
```diff
@@ -2898,6 +3033,12 @@ public partial class RestorableSqlResourceData : Azure.ResourceManager.Models.Re
         public System.Collections.Generic.IReadOnlyList<string> CollectionNames { get { throw null; } }
         public string DatabaseName { get { throw null; } }
     }
+    public partial class Role
```
Comment (fengzhou-msft) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/api/Azure.ResourceManager.CosmosDB.netstandard2.0.cs:3036]: Avoid single-word class name. Rename to a more descriptive name.

**Comment by ashwinisingh01** [comment]
Resolved the comments here: https://github.com/Azure/azure-sdk-for-net/pull/31298

**Pull request closed without merging**